src/details/ArborX_MinimumSpanningTree.hpp

```diff
@@ -30,42 +30,111 @@ namespace ArborX
 namespace Details
 {
 
-struct WeightedEdge
+class WeightedEdge
 {
-  int source;
-  int target;
-  float weight;
+public:
+  unsigned long long _directed_edge = ULLONG_MAX;
+  static constexpr auto inf =
+      KokkosExt::ArithmeticTraits::infinity<float>::value;
+  float _weight = inf;
+
+  static_assert(sizeof(unsigned long long) == 8, "");
+  static constexpr int source_shift = 32;
+  static constexpr int target_shift = 1;
+  static constexpr unsigned long long mask_source =
+      static_cast<unsigned long long>(UINT_MAX >> 1) << source_shift;
+  static constexpr unsigned long long mask_target =
+      static_cast<unsigned long long>(UINT_MAX >> 1) << target_shift;
+  static constexpr unsigned long long mask_undirected = ~((1ull << 63) + 1);
+  static_assert(mask_undirected == mask_source + mask_target, "");
```

Comment:
Do I understand things correctly that there is one bit that is not used? I.e., the first bit (corresponding to the sign bit for `long long`) is not used for anything. You use 62 bits for storing two 31-bit integers, and 1 bit for storing the reverse flag. Correct?

Reply:
Yes